courses/serializers.py

```diff
@@ -329,9 +338,45 @@ def create(self, validated_data):
         )
         return successful_enrollments
 
+    def get_certificate(self, enrollment):
+        """
+        Resolve a certificate for this enrollment if it exists
+        """
+        # When create method is called it returns list object of enrollments
+        if isinstance(enrollment, list):
+            for en in enrollment:
+                self.get_certificate(en)
+        else:
```

Comment:
Let's take a look at what's happening in `create` see line#336. You will notice that we only have one run for this but the enrollment method take a list and returns a list so we convert that single run into list like `[run]`. We get a list in return of create but it is always going to be a list of one element.

Insread, you can pick that enrollment `enrollments[0]` in your case.